adapters/IronSource/IronSourceAdapter/ISMediationManager.m

```diff
@@ -20,425 +20,591 @@
+        return;
+    }
+    
+    if ([self canLoadInterstitialInstance:instanceID]) {
+        [self setInterstitialDelegate:adapterDelegate toInstanceState:GADMAdapterIronSourceInstanceStateLocked];
+        [self addInterstitialDelegate:adapterDelegate forInstanceID:instanceID];
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"ISMediationManager - loadInterstitial  for instance Id %@",
+                instanceID]];
+        [IronSource loadISDemandOnlyInterstitial:instanceID];
+    } else {
+        NSError *error = GADMAdapterIronSourceErrorWithCodeAndDescription(
+                                                                          GADMAdapterIronSourceErrorAdAlreadyLoaded,
+                                                                          @"Instance already loaded. Couldn't load another one in the same time!");
+        [adapterDelegate interstitialDidFailToLoadWithError:error instanceId:instanceID];
+    }
 }
 
 - (void)presentInterstitialAdFromViewController:(nonnull UIViewController *)viewController
                                      instanceID:(nonnull NSString *)instanceID {
-  [GADMAdapterIronSourceUtils
-      onLog:[NSString stringWithFormat:@"ISMediationManager - showInterstitial for instance Id %@",
-                                       instanceID]];
-  [IronSource showISDemandOnlyInterstitial:viewController instanceId:instanceID];
+    [GADMAdapterIronSourceUtils
+     onLog:[NSString stringWithFormat:@"ISMediationManager - showInterstitial for instance Id %@",
+            instanceID]];
+    [IronSource showISDemandOnlyInterstitial:viewController instanceId:instanceID];
+}
+
+- (void)loadBannerAdWithDelegate:(nonnull id<GADMAdapterIronSourceBannerDelegate>)delegate
+                  viewController:(nonnull UIViewController *)viewController
+                      instanceID:(nonnull NSString *)instanceID
+                      bannerSize:(nonnull ISBannerSize*)size {
+    id<GADMAdapterIronSourceBannerDelegate> adapterDelegate = delegate;
+    if (adapterDelegate == nil) {
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"requestBannerAdWithDelegate adapterDelegate is null."]];
+        return;
+    }
+
+        [self setBannerDelegate:adapterDelegate toInstanceState:GADMAdapterIronSourceInstanceStateLocked];
+        [self addBannerDelegate:adapterDelegate forInstanceID:instanceID];
+        
+        
+        [GADMAdapterIronSourceUtils
+         onLog:[NSString
+                stringWithFormat:@"ISMediationManager - loadBanner  for instance Id %@",instanceID]];
+        [IronSource setISDemandOnlyBannerDelegate:self forInstanceId:instanceID];
+        [IronSource loadISDemandOnlyBannerWithInstanceId:instanceID viewController:viewController size:size];
+    
 }
 
 #pragma mark ISDemandOnlyRewardedDelegate
 
 - (void)rewardedVideoAdRewarded:(NSString *)instanceID {
```

Comment:
Can you remind me again why do we need to put all the delegates in this `ISMediationManager` class, rather than in the separate `GADMAdapterIronSource<format>ad` class?